src/richie/apps/search/indexers/courses.py

```diff
@@ -357,15 +357,15 @@ def get_es_document_for_course(cls, course, index=None, action="index"):
 
         # Prepare cover images
         cover_images = {}
-        for cover_image in Picture.objects.filter(
+        for cover in SimplePicture.objects.filter(
             cmsplugin_ptr__placeholder__page=course.extended_object,
             cmsplugin_ptr__placeholder__slot="course_cover",
         ):
-            # Force the image format before computing it
-            cover_image.use_no_cropping = False
-            cover_image.width = COURSES_COVER_IMAGE_WIDTH
-            cover_image.height = COURSES_COVER_IMAGE_HEIGHT
-            cover_images[cover_image.cmsplugin_ptr.language] = cover_image.img_src
+            language = cover.cmsplugin_ptr.language
+            with translation.override(language):
+                cover_images[language] = cover.get_info(
+                    "glimpse", alt=_("{:s} cover").format(titles[language])
```

Comment:
I think this is not ideal as eg. a screen reader would just read the title of the course twice? Once for this and once for the actual title? What's the best practice here?

Reply:
You're right I simplified alts. Full optimization of alts and accessibility needs to be done more seriously in a future PR.